Feature/configuration

grundygolem2 opened the pull request:
yo
it does the do

grundygolem2 committed:
Issue #4 First Pass Monte Carlo result

grundygolem2 committed:
Issue #4 First Pass Monte Carlo result

grundygolem2 committed:
Issue #4 write monte carlo tests
include logging framework + config

grundygolem2 committed:
Issue #4 write monte carlo tests
include logging framework + config

grundygolem2 committed:
Initial dredge configuration

grundygolem2 committed:
Initial dredge configuration

grundygolem2 committed:
remove wrappers

grundygolem2 committed:
remove wrappers

grundygolem2 committed:
Merge branch 'feature/baseDredgeConfig' into feature/monteCarlo

# Conflicts:
#	.idea/modules.xml

grundygolem2 committed:
Begin configuration

grundygolem2 committed:
Begin configuration

grundygolem2 committed:
Get first results

grundygolem2 committed:
Merge remote-tracking branch 'origin/feature/configuration' into feature/configuration

# Conflicts:
#	pom.xml
#	src/main/java/com/github/grundygolem2/configuration/AnalysisConfiguration.java
#	src/main/java/com/github/grundygolem2/montecarlo/MonteCarloExecutor.java
#	src/main/resources/deck-configs/dredge/cards.json
#	src/main/resources/deck-configs/dredge/hands.json

grundygolem2 committed:
Get add postprocessor

grundygolem2 committed:
Update main to support better analysis

grundygolem2 committed:
delete libraries folder

grundygolem2 committed:
extend gitignore to properly ignore libraries folder

clayplumridge commented on pom.xml:
```diff
@@ -61,6 +65,42 @@
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
```
Can we pull test dependencies to the end?

clayplumridge commented on src/main/java/com/github/grundygolem2/Main.java:
```diff
@@ -1,4 +1,40 @@
+//            System.out.println(cards.stream().map(card -> card.getName()).collect(Collectors.toList()) + "->" + tester.test(cards));
+//        }
+
+        double seven = mce.runTest(7, 100000);
```
Definitely want to clean this logic out of Main but it's fine for now

clayplumridge commented on src/main/resources/deck-configs/dredge/hands.json:
```diff
@@ -0,0 +1,146 @@
+      "land"
+    ],
+    "excludesTags": [],
+    "maxKeepSize": 7
```
Only keepable because of the Dredge postprocess ye?

clayplumridge replied:
Actually, postprocess moves us from "keepable hands" to "hands that are likely to work out". Maybe we need to be more selective here.

clayplumridge commented on src/main/resources/deck-configs/dredge/hands.json:
```diff
@@ -0,0 +1,146 @@
+      "land"
+    ],
+    "excludesTags": [],
+    "maxKeepSize": 7
```
Same as above?

clayplumridge commented on src/main/resources/deck-configs/dredge/hands.json:
```diff
@@ -0,0 +1,146 @@
+      "land"
+    ],
+    "excludesTags": [],
+    "maxKeepSize": 7
```
I don't think we keep this with no other stipulations

clayplumridge commented on src/main/resources/deck-configs/dredge/hands.json:
```diff
@@ -0,0 +1,146 @@
+    ],
+    "excludesCards": [],
+    "containsTags": [
+      "land"
```
Not with Dakmor, need a makesRed tag

clayplumridge commented on src/main/resources/deck-configs/dredge/hands.json:
```diff
@@ -0,0 +1,146 @@
+    ],
+    "excludesCards": [],
+    "containsTags": [
+      "land"
```
Not with Dakmor, need red

clayplumridge commented on src/main/resources/deck-configs/dredge/hands.json:
```diff
@@ -0,0 +1,146 @@
+      "dredger"
+    ],
+    "excludesTags": [],
+    "maxKeepSize": 6
```
Not sure we keep this? But it's likely okay. Especially if the dredger is Loam

clayplumridge replied:
Also we should definitely keep "land" "makesGreen" "Life from the Loam" on 5

grundygolem2 committed:
Added SB Cards
Updated hands based on clay's comments
added hand of enabler enabler makesGreen land
added makesRed tag

moved test dependencies to the bottom of the xml

grundygolem2 committed:
update postprocessor to increment depth for every enabler beyond the first

grundygolem2 committed:
Get add postprocessor

grundygolem2 committed:
Get add postprocessor
